src/internal/coredb/projects.go

```diff
@@ -87,8 +106,15 @@ func getProject(ctx context.Context, queryExecer QueryExecer, where string, wher
 	return project, nil
 }
 
-func ListProject(ctx context.Context, db *pachsql.DB) (*ProjectIterator, error) {
-	rows, err := db.QueryxContext(ctx, "SELECT name, description, created_at FROM core.projects")
+func ListProject(ctx context.Context, db *pachsql.DB, option ...ListProjectOption) (*ProjectIterator, error) {
```

Comment:
I don't think it make sense to combine pagination and iterators. If you don't have pagination, then the results could be huge, so iterators make sense, but with pagination, I think it's better to just return the results as a slice. Basically, I think `ProjectIterator` should be able to "transparently" iterate through all projects (with or without pagination).

So I propose we have two ways to iterate:
- `ListAllProjects() (*ProjectIterator, error)`, behind the scenes, this could use the paged ListProject, which means for now, you don't have to implement this.
- `ListProject() ([]*ProjectInfo, error)`

Reply:
Gotcha, I think maybe its better to just stick with the iterator then since that's already implemented. I can supply the other paginated approach for transactions